astroML/stats/_binned_statistic.py

```diff
@@ -338,7 +336,7 @@ def binned_statistic_dd(sample, values, statistic='mean',
     elif callable(statistic):
         try:
             null = statistic([])
-        except:
+        except Exception:
```

Comment:
would be nice to be more specific here

Reply:
This one is very difficult to fix. The `statistics` is a `string or callable, optional` but if the given `statistics` is a user-defined callable then it can throw any error. [The documentation](https://github.com/astroML/astroML/blob/master/astroML/stats/_binned_statistic.py#L34) does not specify what error should `statistic` raise. Perhaps we need to insist `ValueError` has to be returned for empty array and then test for it explicitly and raise a different error if not?

If not PEP8 prefers `except Exception` instead of a bare `except`.